src/script/sign.h

```diff
@@ -8,19 +8,52 @@
 
 #include "script/interpreter.h"
 
+class CKeyID;
 class CKeyStore;
 class CScript;
 class CTransaction;
 
 struct CMutableTransaction;
 
+/** Virtual base class for signature creators. */
+class BaseSignatureCreator {
+protected:
+    const CKeyStore& keystore;
```

Comment:
Let's follow #5710 and make this (as well as the argument to BaseSignatureCreator) a pointer instead of a reference. This will avoid any possibility of bugs caused by storing a reference to a temporary, implicit conversion.